packages/trace-viewer/src/ui/errorsTab.tsx

```diff
@@ -86,15 +89,54 @@ export const ErrorsTab: React.FunctionComponent<{
   wallTime: number,
   sdkLanguage: Language,
   revealInSource: (error: modelUtil.ErrorDescription) => void,
-}> = ({ errorsModel, model, sdkLanguage, revealInSource, wallTime }) => {
+  testRunMetadata: MetadataWithCommitInfo | undefined,
+}> = ({ errorsModel, model, sdkLanguage, revealInSource, wallTime, testRunMetadata }) => {
   const errorContext = useAsyncMemo(async () => {
     const attachment = model?.attachments.find(a => a.name === 'error-context');
     if (!attachment)
       return;
     return await fetch(attachmentURL(attachment)).then(r => r.text());
   }, [model], undefined);
 
-  const prompt = fixTestInstructions + (errorContext ?? ''); // TODO in next PR: enrich with test location, error details and source code, similar to errorContext.ts
+  const prompt = useAsyncMemo(
+      () => copyPrompt(
+          model?.title ?? '',
+          model?.errorDescriptors ?? [],
+          testRunMetadata,
+          errorContext,
+          async error => {
```

Comment:
- Do we want to try/catch this block?
- Let's extract a named helper function instead of anonymous arrow function.